server/model/board.go

```diff
@@ -19,6 +19,10 @@ type Board struct {
 	// required: true
 	ID string `json:"id"`
 
+	// The TemplateID for the board
```

Comment:
I added this new field, I'm not sure about the name, because looks like is for storing the template that created the board, but in reality is set only in built in templates for tracking it. Maybe it should be something like `BuiltinTemplateId` or `BlueprintTemplateId` or `BuildinTemplateTrackingId`. @harshilsharma63 @mgdelacroix what do you think?